push-to-s3-and-invalidate-cloudfront/action.yml

```
@@ -0,0 +1,150 @@
+
+runs:
+  using: composite
+  steps:
+    - name: Compute the S3 directory to upload to
+      shell: bash
+      id: upload
+      run: |
+        if [[ "${{ inputs.s3-dir-to-upload-to }}" == "pr-previews" ]]
+        then
+          dir_to_upload_to="${{ inputs.s3-dir-to-upload-to }}/PR-${{ inputs.pull-request-number }}"
+        elif [[ "${{ inputs.s3-dir-to-upload-to }}" == "/" ]]
+        then
+          dir_to_upload_to=""
+        else
+          dir_to_upload_to="${{ inputs.s3-dir-to-upload-to }}"
+        fi
+        echo "dir_to_upload_to=$dir_to_upload_to" >> $GITHUB_OUTPUT
+
+    - name: Compute the S3 directory to delete from
+      shell: bash
+      id: delete
+      run: |
+        if [[ "${{ inputs.s3-dir-to-delete-from }}" == "pr-previews" ]]
+        then
+          dir_to_delete_from="${{ inputs.s3-dir-to-delete-from }}/PR-${{ inputs.pull-request-number }}"
+        elif [[ "${{ inputs.s3-dir-to-delete-from }}" == "/" ]]
+        then
+          dir_to_delete_from=""
+        else
+          dir_to_delete_from="${{ inputs.s3-dir-to-delete-from }}"
+        fi
+        echo "dir_to_delete_from=$dir_to_delete_from" >> $GITHUB_OUTPUT
+
+    - name: Compute files to exclude
+      shell: bash
+      id: exclude
+      run: |
+        files_to_exlude=""
+        if [[ "${{ inputs.s3-files-to-exclude }}" == *,* ]]
+        then
+          IFS=','
+          for i in `echo "${{ inputs.s3-files-to-exclude }}"`
+          do
+            files_to_exlude="$files_to_exlude --exclude \"$i\""
+          done
+          unset IFS
+        elif [[ "${{ inputs.s3-files-to-exclude }}" == "pr-previews" ]] || [[ "${{ inputs.s3-files-to-exclude }}" != "" ]]
+        then
+          files_to_exlude="--exclude \"${{ inputs.s3-files-to-exclude }}\""
+        fi
+        echo "files_to_exlude=$files_to_exlude" >> $GITHUB_OUTPUT
+
+    - name: Upload the files to S3
+      shell: bash
+      if: inputs.s3-action == 'upload' || (inputs.s3-action == '' && github.event_name == 'pull_request' && contains(fromJson('["opened", "synchronize"]'), github.event.action))
+      env:
+        AWS_REGION: ${{ inputs.aws-region }}
+        AWS_ACCESS_KEY_ID: ${{ inputs.aws-access-key-id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.aws-secret-access-key }}
+      run: |
+        aws s3 sync ${{ inputs.build-directory }}/ s3://${{ inputs.s3-bucket }}/${{ steps.upload.outputs.dir_to_upload_to }} ${{ steps.exclude.outputs.files_to_exlude }}
+
+    - name: Delete the files on S3
+      shell: bash
+      if: inputs.s3-action == 'delete' || (inputs.s3-action == '' && github.event_name == 'pull_request' && github.event.action == 'closed')
+      env:
+        AWS_REGION: ${{ inputs.aws-region }}
+        AWS_ACCESS_KEY_ID: ${{ inputs.aws-access-key-id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.aws-secret-access-key }}
+      run: aws s3 rm s3://${{ inputs.s3-bucket }}/${{ steps.upload.outputs.dir_to_delete_from }}/ --recursive
```

So going with the above suggestion on the directory steps, this would get updated to:
```suggestion
      run: aws s3 rm s3://${{ inputs.s3-bucket }}/${{ steps.s3_directory.outputs.dir_name }}/ --recursive
```

.. also, we want to exclude files in this step too no?

---

Hmm, I've never used it for deletion but doesn't hurt to have it here. Added it!